lab-bundle-spec.md

```diff
@@ -308,20 +308,140 @@ environment:
 
-Lab bundles will provisionally support the JSON representation of Activity Tracking currently used in the Qwiklabs web interface. The JSON definition should be stored in file separately from (and referenced directly in) `qwiklabs.yaml`.
+A lab has an Assessment, which in turn contains the Steps (checkpoints).
 
-```yaml
-entity_type: Lab
+#### Assessment
 
-...
+attribute          | required | type    | notes
+-------------------| -------- | --------| --------------------------------------
+passing_percentage | ✓        | integer | The percentage of total points the student must achieve to "pass" the lab.
+steps              | ✓        | array   | An array of [Steps](#steps)
 
-activity_tracking: ./assessment/activity_tracking.json
+```yml
+assessment:
+  passing_percentage: 75
+  steps: ...
 ```
 
-> **Note:** Support for this format should be considered deprecated.
->
-> Further work will be done to define a new DSL for expressing Activity Tracking logic. Exploratory sketches of this DSL can be found in the `./examples` directory of this repository.
+#### Steps
+
+attribute        | required | type                       | notes
+-----------------| -------- | ---------------------------| --------------------------------------
+title            | ✓        | locale dictionary          |
+maximum_score    | ✓        | integer                    | The maximum number of points this step can award.
+student_messages | ✓        | array of objects           | The keys are how the messages will be referenced in the code, and the values are locale dictionaries.
+services         | ✓        | array of resource services | An array of services that will be used in the code block. Each resource type specifies a set of allowed services.
+code             | ✓        | string                     | Code to be executed. See [below](#code) for more information.
+
+```yml
+assessment:
+  passing_percentage: 75
+  steps:
+    - title:
+        locales:
+          en: Create a Cloud Storage bucket
+          es: Crear un depósito de almacenamiento en la nube
+      maximum_score: 5
+      student_messages:
+        - success:
+            locales:
+              en:
+                Great job! You created the bucket!
+              es:
+                ¡Gran trabajo! ¡Creaste el cubo!
+        - bucket_missing:
+            locales:
+              en:
+                Oops! No bucket found.
+              es:
+                ¡Uy! No se ha encontrado el cubo.
+        - bucket_misconfigured:
+            locales:
+              en:
+                Hmm. The bucket is there, but it is misconfigured.
+              es:
+                Hmm. El cubo está allí, pero está mal configurado.
+      services:
```

Comment:
I'd like creators to specify a per-step field and I don't know what to call it but basically I'd like creators to specify what granularity their step applies to:

* `zone`: Run this in every GCP zone (e.g. checking if a GCE instance was created in that zone)
* `region`: Run this in every GCP region (e.g. checking if a regional GKE cluster was made in that region)
* `multiregion`: Run this once for US, EU, and Asia (e.g. checking for multi-regional GCS buckets or some newer BigQuery data locality stuff)
* `global`: Run this function once (e.g. checking IAM configuration)

Reply:
interesting. i don't think we want to run any of this stuff in every zone, and even running it in every region has caused us headaches. i would much rather keep everything "global" until we have a really compelling use case and/or rearchitect things to make "run n times" much easier.